mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java

```diff
@@ -249,6 +258,33 @@ public void messagesRetrievedUsingFetchTypeBodyShouldHaveBodyDataLoaded() throws
     }
 
     @Test
+    public void messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsIdsLoadedWhenNoAttachment() throws MailboxException, IOException{
+        saveMessages();
+        MessageMapper.FetchType fetchType = MessageMapper.FetchType.Full;
+        Iterator<MailboxMessage> retrievedMessageIterator = messageMapper.findInMailbox(benwaInboxMailbox, MessageRange.one(message1.getUid()), fetchType, LIMIT);
+        MessageAssert.assertThat(retrievedMessageIterator.next()).isEqualTo(message1, fetchType);
+        assertThat(retrievedMessageIterator).isEmpty();
+    }
+
+    @Test
+    public void messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsIdsLoadedWhenOneAttachment() throws MailboxException, IOException{
+        saveMessages();
+        MessageMapper.FetchType fetchType = MessageMapper.FetchType.Full;
+        Iterator<MailboxMessage> retrievedMessageIterator = messageMapper.findInMailbox(attachmentsMailbox, MessageRange.one(message7With1Attachment.getUid()), fetchType, LIMIT);
```

Comment:
Is it only true with Fetch type Full ?

For which type of Fetch Type could the attachement Ids not be retrieved ?